examples/wavenet.cc

```diff
@@ -0,0 +1,229 @@
+            where out in 0:IN # necessary because r_in gets into unresolved min/max
+}
+  )TC";
+  tc::ATenCompilationUnit<tc::CudaTcExecutor> atCompl;
+  atCompl.define(tc);
+
+  // 2. Allocate tensors with random data.
+  at::Tensor weight0 = at::CUDA(at::kFloat).rand({128, 64, 2});
+  at::Tensor bias0 = at::CUDA(at::kFloat).rand({128});
+  at::Tensor data0 = at::CUDA(at::kFloat).rand({1, 64, 2});
+  at::Tensor res_weight0 = at::CUDA(at::kFloat).rand({64, 64});
+  at::Tensor res_bias0 = at::CUDA(at::kFloat).rand({64});
+  at::Tensor skip_weight0 = at::CUDA(at::kFloat).rand({256, 64});
+  at::Tensor skip_bias0 = at::CUDA(at::kFloat).rand({256});
+
+  at::Tensor weight1 = at::CUDA(at::kFloat).rand({128, 64, 2});
+  at::Tensor bias1 = at::CUDA(at::kFloat).rand({128});
+  at::Tensor data1 = at::CUDA(at::kFloat).rand({1, 64, 2});
+  at::Tensor res_weight1 = at::CUDA(at::kFloat).rand({64, 64});
+  at::Tensor res_bias1 = at::CUDA(at::kFloat).rand({64});
+  at::Tensor skip_weight1 = at::CUDA(at::kFloat).rand({256, 64});
+  at::Tensor skip_bias1 = at::CUDA(at::kFloat).rand({256});
+
+  // 3. Run autotuning with evolutionary search starting from a naive option.
+  auto naiveOptions = tc::CudaMappingOptions::makeNaiveCudaMappingOptions();
+  tc::autotune::GeneticAutotunerATen geneticAutotuneATen(tc);
+  auto bestOption = geneticAutotuneATen.tune(
+      FLAGS_tuner_proto,
+      "wavenet2layers",
+      {weight0,
+       bias0,
+       data0,
+       res_weight0,
+       res_bias0,
+       skip_weight0,
+       skip_bias0,
+       weight1,
+       bias1,
+       data1,
+       res_weight1,
+       res_bias1,
+       skip_weight1,
+       skip_bias1},
+      naiveOptions);
+
+  // 4. Compile and run the TC with the best option.
+  // Outputs get allocated; could also be pre-allocated and passed.
+  auto handle = atCompl.compile(
+      "wavenet2layers",
+      {weight0,
+       bias0,
+       data0,
+       res_weight0,
+       res_bias0,
+       skip_weight0,
+       skip_bias0,
+       weight1,
+       bias1,
+       data1,
+       res_weight1,
+       res_bias1,
+       skip_weight1,
+       skip_bias1},
+      bestOption.getValue());
+  std::vector<at::Tensor> outputs;
+  auto duration = atCompl.run(
+      "wavenet2layers",
+      {weight0,
+       bias0,
+       data0,
+       res_weight0,
+       res_bias0,
+       skip_weight0,
+       skip_bias0,
+       weight1,
+       bias1,
+       data1,
+       res_weight1,
+       res_bias1,
+       skip_weight1,
+       skip_bias1},
+      outputs,
+      handle,
+      true);
+  std::cout
+      << "wavenet2layers size weight0: " << weight0.sizes() << " ran in: "
+      << std::chrono::duration_cast<std::chrono::microseconds>(duration).count()
+      << "us\n";
+
+  // 5. TODO: the following represent reasonable initialization
+  // operations. Port them from PyTorch if we want to look at precision in the
+  // future.
+  //   weight0 = 5*(torch.cuda.FloatTensor(128,64,2).uniform_() - 0.5)
```

Comment:
These seem pretty straightforward to translate, was there any particular problem?